### ISSUE TITLE
Increase database engine versions / make tt

### ISSUE THREAD

**Issue by aspark21** [moodlehq/moodle-docker#170]
https://github.com/moodlehq/moodle-docker/blob/master/db.mariadb.yml#L8
https://github.com/moodlehq/moodle-docker/blob/master/db.mysql.yml#L8

**Comment by stronk7** [comment]
Yeah, we have a similar request also for [moodle-ci-runner](https://github.com/moodlehq/moodle-ci-runner), that runs all our CI jobs, see [MDLSITE-6376](https://tracker.moodle.org/browse/MDLSITE-6376).

Surely both (they are different, but "parallel" enough) could be achieved in a similar way.

A) For some reason what I had been thinking was, more or less:

1. Create new, unique, `MOODLE_DOCKER_DB_VERSION` (say, 8, or 13, or 5.7.12...) variable.
2. If not set, each database will default to current values.
3. If set, then use it as tag for `MOODLE_DOCKER_DB` image (mysql, postgres...) to use.
4. Always look for `db.MOODLE_DOCKER_DB.MOODLE_DOCKER_DB_VERSION`  (db.mysql8.yml, db.oracle19.yml...) and use them if they exist.
5. Fallback to un-versioned `db.MOODLE_DOCKER_DB` (db.mysql.yml, db.oracle.yml...) if versioned doesn't exist.

I'm aware it's not the unique alternative. Maybe we could also:

B) Do everything with existing `MOODLE_DOCKER_DB`, making it to accept database:tag format.
C) Make a different `MOODLE_DOCKER_DB_VERSION` variable for each database.
D) ...

It would be really interesting to see if we can come with an agreement and get this implemented, yes!

Ciao :-)

**Comment by aspark21** [comment]
Got a version of it working - https://github.com/aspark21/moodle-docker/blob/mysql8/db.mysql8.yml
Had to remove 2 settings deprecated in 8 and add one for auth - https://github.com/moodlehq/moodle-docker/compare/master...aspark21:100c005241743552d2fe2a08157557b5c5f4fb54

For that reason, we're going to need a separate image.

Just verifying it runs properly and I'll submit a MR with that first step.


Having a look at moodle-ci-runner, the version change would need to take place here:
https://github.com/moodlehq/moodle-ci-runner/blob/master/runner/master/run.sh#L283
https://github.com/moodlehq/moodle-ci-runner/blob/master/runner/master/run.sh#L299

The deprecated parameters were already removed to enable mariadb
https://github.com/moodlehq/moodle-ci-runner/commit/21dfcbc2bbbe8c5e5c9385502b7452b51cbf8f98

So the main thing is wether or not --default-authentication-plugin=mysql_native_password is required. 
This looks to have remained the default in mariadb & was only changed in mysql8.0
The setting itself has been around for a while so shouldn't be an issue to add that setting to the file for all versions&forks to use.

**Comment by stronk7** [comment]
Just for reference and similarities... we already added support to DB tags to `moodle-ci-runner` @ https://github.com/moodlehq/moodle-ci-runner/pull/78.

**Comment by skodak** [comment]
I have added pull https://github.com/moodlehq/moodle-docker/pull/225 which allows you to specify database version

**Comment by stronk7** [comment]
Hi @aspark21 ,

closing this in favour of #225, that is going to land soon.

Ciao :)
